cmd/daemon/start.go

```diff
@@ -7,6 +7,7 @@ import (
 	"path/filepath"
 	"time"
 
+	flock "github.com/gofrs/flock"
```

Comment:
I think instead of adding a new dependency, you can use sync.RWMutex, which gives us a write locker based on the file system mechanism. For a file, you could only have one writer, and I believe this will solve the issue.

Reply:
Based on my understanding, sync.RWMutex is a lock for synchronization between two threads, and it doesn't have the capability to accomplish this. 
> For a file, you could only have one writer

`flock` is designed to handle such tasks

Reply:
I meant that you can use these solutions, etc., to prevent new dependencies.

Reply:
The flock is a file-locking system @amirvalhalla . Not process locking